src/core/pre.js

```diff
@@ -6,3 +6,39 @@ API.version = "0.24.0.dev0";
 Module.hiwire = Hiwire;
 const getTypeTag = (x) => Object.prototype.toString.call(x);
 API.getTypeTag = getTypeTag;
+
+/**
+ * Safe property check
+ *
+ * Observe whether a property exists or not without invoking getters.
+ * Should never throw as long as arguments have the correct types.
+ *
+ * obj: an object
+ * prop: a string or symbol
+ */
+function hasProperty(obj, prop) {
+  while (obj) {
+    if (Object.getOwnPropertyDescriptor(obj, prop)) {
```

Comment:
Why not use hasOwn here?

Reply:
Too recent:
https://caniuse.com/?search=hasOwn